### DISCUSSION TITLE
How to infer with the created model

### DISCUSSION THREAD

**Discussion by Shin-ichi-Takayama** [speechbrain/speechbrain#1833] [Q&A] [answered]
I have trained the model on my own data using the recipe WSJ02Mix.
I want to infer using the model I have created.
Could you please tell me how to do the inference?

**Comment by anautsch** [comment]
Hi @Shin-ichi-Takayama I'm not an expert with this recipe - but on HuggingFace, there are two pretrained models using WSJ02Mix
https://huggingface.co/speechbrain/sepformer-wsj02mix
https://huggingface.co/speechbrain/sepformer-wsj03mix

Their model cards demonstrate how to use such a pretrained model with `speechbrain.pretrained` interfaces e.g.:
```python
from speechbrain.pretrained import SepformerSeparation as separator
import torchaudio

model = separator.from_hparams(source="speechbrain/sepformer-wsj02mix", savedir='pretrained_models/sepformer-wsj02mix')

# for custom file, change path
est_sources = model.separate_file(path='speechbrain/sepformer-wsj02mix/test_mixture.wav') 

torchaudio.save("source1hat.wav", est_sources[:, :, 0].detach().cpu(), 8000)
torchaudio.save("source2hat.wav", est_sources[:, :, 1].detach().cpu(), 8000)
```

Simply replace the `source="speechbrain/sepformer-wsj02mix"` with the local path to your model.

**Reply by Shin-ichi-Takayama** [reply]
Hi @anautsch.
Sorry for the late reply.
By setting 'test_only' to True in the configuration yaml file, only the inference could be run.
Thank you for your response.

**Comment by Shin-ichi-Takayama** [comment, marked as answer]
The following methods have been used to solve the problem.
Set `test_only` to True in the configuration yaml file.
Run `train.py` after the change.
